File: src/main/java/org/library/BorrowedBooks.java
package org.library;

import java.util.List;
import java.util.ArrayList;

public class BorrowedBooks {
    List<Book> books;

    public BorrowedBooks(){
        books = new ArrayList<>();
    }

    public Book getBookByTitle(String bookTitle){
        for (int i=0; i<size(); ++i){
            Book b = books.get(i);
            if (b.getTitle().equals(bookTitle))
                return b;
        }
        return null;
    }

    public void addBook(Book b){
        books.add(b);
    }

    public void removeBook(String bookTitle){
        for (int i=0; i<size(); ++i){
            Book b = books.get(i);
            if (b.getTitle().equals(bookTitle)){
                books.remove(i);
                return;
            }
        }
    }

    public List<Book> getBorrowedBooksList(){
        return books;
    }

    public int size(){ return books.size(); }
}

File: main/BorrowedBooks.js
export default class BorrowedBooks {
  constructor() {
    this.books = [];
  }

  getBookByTitle(bookTitle) {
    for (let i = 0; i < this.size(); ++i) {
      const b = this.books[i];
      const title = typeof b.getTitle === "function" ? b.getTitle() : b.title;
      if (title === bookTitle) return b;
    }
    return null;
  }

  addBook(b) {
    this.books.push(b);
  }

  removeBook(bookTitle) {
    for (let i = 0; i < this.size(); ++i) {
      const b = this.books[i];
      const title = typeof b.getTitle === "function" ? b.getTitle() : b.title;
      if (title === bookTitle) {
        this.books.splice(i, 1);
        return;
      }
    }
  }

  getBorrowedBooksList() {
    return this.books;
  }

  size() {
    return this.books.length;
  }
}
